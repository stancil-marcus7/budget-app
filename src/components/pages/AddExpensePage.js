import React from 'react';
import ExpenseForm from '../ExpenseForm'
import {connect} from 'react-redux' 
import {startAddExpense} from '../../actions/expenses'
export class AddExpensePage extends React.Component{
  onSubmit = (expense) => {
    this.props.startAddExpense(expense);
    //This will switch us over to the home page
    this.props.history.push('/');
  };

  render() {
    return (
      <div>
          <div className="page-header">
            <div className="content-container">
              <h1 className="page-header__title">Add Expense</h1>
            </div>
          </div>
          <div className="content-container">
            <ExpenseForm
                  button="Add Expense"
                  onSubmit={this.onSubmit}/>
          </div>
      </div>
    )
  }
}

//Allows you to create props out of dispatch
const mapDispatchToProps = (dispatch) => ({
    startAddExpense: (expense) => dispatch(startAddExpense(expense))
})

export default connect(undefined, mapDispatchToProps)(AddExpensePage);
