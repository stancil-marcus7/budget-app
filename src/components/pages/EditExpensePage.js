import React from 'react';
import { connect } from 'react-redux';
import ExpenseForm from '../ExpenseForm'
import { editExpense } from '../../actions/expenses'
import { removeExpense } from '../../actions/expenses'

export class EditExpensePage extends React.Component {
render(){
  return (
    <div>
      <ExpenseForm
        expense={this.props.expense}
        onSubmit={expense => {
          this.props.editExpense(this.props.expense.id, expense);
          this.props.history.push('/');
        }}
        />
        <button onClick={expense => {
            this.props.removeExpense({id: this.props.expense.id})
            this.props.history.push('/')
          }}
            
        >Remove</button>
    </div>
  )
} 
}

const mapStateToProps = (state, props) => {
  return {
    expense: state.expenses.find(expense => expense.id === props.match.params.id
    )
  }
}

const mapDispatchToProps = (dispatch) => ({
  //need all params from action
  editExpense: (id, expense) => dispatch(editExpense(id, expense)),
  removeExpense: (expense) => dispatch(removeExpense({id: expense.id}))
})
export default connect(mapStateToProps,mapDispatchToProps)(EditExpensePage);
