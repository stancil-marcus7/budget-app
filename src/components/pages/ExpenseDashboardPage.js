import React from 'react';
import getExpenseTotal from "../../actions/expenses-total";
import { connect } from 'react-redux';
import numeral from 'numeral';
import selectExpenses from '../../selectors/expenses'

export class ExpenseDashboardPage extends React.Component{
    render(){
        return (
            <div>
                <p>Viewing {this.props.expenseCount} {this.props.expenseCount > 1 || this.props.expenseCount === 0 ? "expenses " : "expense "} totalling {numeral(this.props.expenseTotal/100).format('$0,0.00')}</p>
            </div>
        )
    }
}

const mapStateToProps = (state) => {
    return {
        expenseCount: selectExpenses(state.expenses, state.filters).length,
        expenseTotal: getExpenseTotal(selectExpenses(state.expenses, state.filters))
    }
}



export default connect(mapStateToProps)(ExpenseDashboardPage); 