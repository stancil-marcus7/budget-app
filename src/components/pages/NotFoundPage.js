import React from 'react';
import {NavLink} from 'react-router-dom'

const NotFoundPage = () => (
  <div>
      404! - <NavLink to="/">Go Home!</NavLink>
  </div>
)

export default NotFoundPage;
