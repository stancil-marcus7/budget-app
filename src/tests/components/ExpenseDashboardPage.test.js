import {ExpenseDashboardPage} from "../../components/pages/ExpenseDashboardPage";
import { shallow } from 'enzyme';
import React from 'react';

test(  `should render with all the expsneses`, () => {
    const wrapper = shallow(<ExpenseDashboardPage expenseCount={3} expenseTotal={10}/>)
    expect(wrapper).toMatchSnapshot();
})
  
test(  `should render no expenses, when there are no expenses`, () => {
    const wrapper = shallow(<ExpenseDashboardPage expenseCount={0} expenseTotal={0}/>)
    expect(wrapper).toMatchSnapshot();
})