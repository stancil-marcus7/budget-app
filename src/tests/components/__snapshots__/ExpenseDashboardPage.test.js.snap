// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render no expenses, when there are no expenses 1`] = `
<div>
  <p>
    Viewing 
    0
     
    expenses 
     totalling 
    $0.00
  </p>
</div>
`;

exports[`should render with all the expsneses 1`] = `
<div>
  <p>
    Viewing 
    3
     
    expenses 
     totalling 
    $0.10
  </p>
</div>
`;
